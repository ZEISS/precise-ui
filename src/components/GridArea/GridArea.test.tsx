import * as React from 'react';
import * as enzyme from 'enzyme';
import { GridArea } from './';

describe('<GridArea />', () => {
  it('should render an empty <GridArea> component', () => {
    const wrapper = enzyme.shallow(<GridArea />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <GridArea> component with children', () => {
    const wrapper = enzyme.shallow(
      <GridArea>
        <div>Test1</div>
        <div>Test2</div>
      </GridArea>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <GridArea> component with column and row span', () => {
    const wrapper = enzyme.shallow(<GridArea colSpan={2} rowSpan={2} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <GridArea> component with column and row span', () => {
    const wrapper = enzyme.shallow(<GridArea colSpan={2} rowSpan={2} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <GridArea> component with column and row index', () => {
    const wrapper = enzyme.shallow(<GridArea column={2} row={2} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <GridArea> component with column index and row span including children', () => {
    const wrapper = enzyme.shallow(
      <GridArea column={2} colSpan={3}>
        Test
      </GridArea>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
