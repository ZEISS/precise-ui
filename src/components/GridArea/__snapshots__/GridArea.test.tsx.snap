// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridArea /> should render a <GridArea> component with children 1`] = `
<styled.div
  onMouseDown={[Function]}
  onTouchStartCapture={[Function]}
>
  <div>
    Test1
  </div>
  <div>
    Test2
  </div>
</styled.div>
`;

exports[`<GridArea /> should render a <GridArea> component with column and row index 1`] = `
<styled.div
  onMouseDown={[Function]}
  onTouchStartCapture={[Function]}
/>
`;

exports[`<GridArea /> should render a <GridArea> component with column and row span 1`] = `
<styled.div
  onMouseDown={[Function]}
  onTouchStartCapture={[Function]}
/>
`;

exports[`<GridArea /> should render a <GridArea> component with column and row span 2`] = `
<styled.div
  onMouseDown={[Function]}
  onTouchStartCapture={[Function]}
/>
`;

exports[`<GridArea /> should render a <GridArea> component with column index and row span including children 1`] = `
<styled.div
  onMouseDown={[Function]}
  onTouchStartCapture={[Function]}
>
  Test
</styled.div>
`;

exports[`<GridArea /> should render an empty <GridArea> component 1`] = `
<styled.div
  onMouseDown={[Function]}
  onTouchStartCapture={[Function]}
/>
`;
